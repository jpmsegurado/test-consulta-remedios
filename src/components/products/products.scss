.products {
  width: 100%;
  display: flex;
  flex-wrap: wrap;

  > .product-wrapper {
    width: calc(100% / 3);
    padding: 15px;
    box-sizing: border-box;
  }
}