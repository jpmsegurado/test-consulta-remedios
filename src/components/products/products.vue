<template>
  <div class="products">
    <div class="product-wrapper" v-for="product in products" :key="product.id">
      <product :product-info="product" />
    </div>
  </div>
</template>

<script>
import product from "../product/product";
import { mapGetters, mapActions } from "vuex";

export default {
  components: {
    product
  },
  computed: {
    ...mapGetters({
      products: "PRODUCT/ALL"
    })
  },
  methods: {
    ...mapActions({
      loadProducts: "PRODUCT/LOAD_PRODUCTS"
    })
  },
  mounted() {
    this.loadProducts();
  }
};
</script>

<style lang="scss" scoped src="./products.scss" />
