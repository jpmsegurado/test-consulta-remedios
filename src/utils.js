export function formatMoney(value, decimal = 2) {
  const formatter = new Intl.NumberFormat("pt-BR", {
    style: "currency",
    currency: "BRL",
    minimumFractionDigits: decimal
  });

  return formatter.format(value);
}
