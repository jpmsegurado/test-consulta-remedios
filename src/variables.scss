$product-bg: #EEEEEE;
$product-name: #746A6A;
$product-price: #3486E6;
$cart-border: #E1E1E1;
$cart-text: #746A6A;
$item-cart-name: #7F7575;
$white: #fff;
$cart-button: #54A3FF;
$select-border: #E1E1E1;

.fade-enter-active, .fade-leave-active {
  transition: opacity .3s;
}
.fade-enter, .fade-leave-to {
  opacity: 0;
}