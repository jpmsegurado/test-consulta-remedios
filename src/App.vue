<template>
  <div id="app">
    <div class="row">
      <div class="products-wrapper">
        <h2>Games</h2>
        <mySelect v-model="selected" :value="selected" :list="options" />
        <products :selectedOption="options[selected]" />
      </div>

      <div class="shopping-cart-wrapper">
        <shopping-cart />
      </div>
    </div>
  </div>
</template>

<script>
import products from "./components/products/products";
import shoppingCart from "./components/shopping-cart/shopping-cart";
import mySelect from "./components/select/select";

export default {
  name: "app",
  components: {
    products,
    shoppingCart,
    mySelect
  },
  data() {
    return {
      options: [
        { label: "Mais populares", prop: "score" },
        { label: "Mais baratos", prop: "price" },
        { label: "Ordem Alfab√©tica", prop: "name" }
      ],
      selected: 0
    };
  }
};
</script>

<style lang="scss">
@import url("https://fonts.googleapis.com/css?family=Open+Sans");
body {
  padding: 0;
  margin: 0;
}
#app {
  font-family: "Open Sans";
  padding: 42px 70px;

  > .row {
    display: flex;
    width: 100%;
    flex-wrap: wrap;

    .products-wrapper {
      width: 75%;

      > h2 {
        font-size: 48px;
        margin-top: 0;
        margin-bottom: 44px;
        width: calc(100% - 277px);
        display: inline-block;
      }

      > .select {
        width: 262px;
        margin-right: 15px;
      }
    }

    .shopping-cart-wrapper {
      width: 25%;
    }
  }
}
</style>
