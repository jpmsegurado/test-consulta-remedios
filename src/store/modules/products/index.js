import products from "../../../assets/products.json";

const state = {
  products: []
};

const getters = {
  ALL: state => state.products
};

const actions = {
  LOAD_PRODUCTS: ({ commit }) => {
    return Promise.resolve(commit("SET_PRODUCTS", products));
  }
};

const mutations = {
  SET_PRODUCTS: (state, products) => {
    state.products = products;
  }
};

export default {
  namespaced: true,
  state,
  getters,
  actions,
  mutations
};
