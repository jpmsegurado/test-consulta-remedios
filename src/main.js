import Vue from "vue";
import App from "./App.vue";
import store from "./store/store";
import { formatMoney } from "./utils";

Vue.config.productionTip = false;

Vue.prototype.formatMoney = formatMoney;

new Vue({
  store,
  render: h => h(App)
}).$mount("#app");
