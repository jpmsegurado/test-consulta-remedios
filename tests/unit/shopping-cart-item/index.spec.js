import { shallowMount } from "@vue/test-utils";
import shoppingCartItem from "@/components/shopping-cart-item/shopping-cart-item.vue";
import products from "@/assets/products.json";
import { formatMoney } from "../../../src/utils";
import groupBy from "lodash/fp/groupBy";
import sumBy from "lodash/fp/sumBy";

function groupProducts(products) {
  const grouped = groupBy(item => item.id)(products);
  const ids = Object.keys(grouped);
  const values = Object.values(grouped);

  return ids.map((id, index) => ({
    id: Number(id),
    products: values[index],
    name: values[index][0].name,
    image: values[index][0].image,
    total: sumBy(product => product.price)(values[index])
  }));
}

const shoppingCartProducts = [];

shoppingCartProducts.push(products[0]);

describe("shopping-cart-item.vue", () => {
  it("render shopping cart item", () => {
    const productInfo = groupProducts(shoppingCartProducts)[0];
    const wrapper = shallowMount(shoppingCartItem, {
      propsData: { productInfo },
      mocks: {
        formatMoney
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
  it("render shopping cart item with repeated item", () => {
    shoppingCartProducts.push(products[0]);
    const productInfo = groupProducts(shoppingCartProducts)[0];
    const wrapper = shallowMount(shoppingCartItem, {
      propsData: { productInfo },
      mocks: {
        formatMoney
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
