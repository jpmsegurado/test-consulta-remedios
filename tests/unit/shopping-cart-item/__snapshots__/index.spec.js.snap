// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shopping-cart-item.vue render shopping cart item 1`] = `
<div class="shopping-cart-item">
  <div class="item-image"><img src="" alt="Fifa 18"></div>
  <div class="item-info">
    <p>
      Super Mario Odyssey

    </p>
    <p>R$&nbsp;197.88</p>
  </div> <button>X</button>
</div>
`;

exports[`shopping-cart-item.vue render shopping cart item with repeated item 1`] = `
<div class="shopping-cart-item">
  <div class="item-image"><img src="" alt="Fifa 18"></div>
  <div class="item-info">
    <p>
      Super Mario Odyssey
      (2x)
    </p>
    <p>R$&nbsp;395.76</p>
  </div> <button>X</button>
</div>
`;
